x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/action/filter/ShardBulkInferenceActionFilterTests.java

```diff
@@ -92,21 +91,10 @@ public void tearDownThreadPool() throws Exception {
         terminate(threadPool);
     }
 
-    private IndexVersion getRandomIndexVersion() {
-        return randomFrom(
-            IndexVersionUtils.randomVersionBetween(
-                random(),
-                IndexVersions.SEMANTIC_TEXT_FIELD_TYPE,
-                IndexVersionUtils.getPreviousVersion(IndexVersions.INFERENCE_METADATA_FIELDS)
-            ),
-            IndexVersionUtils.randomVersionBetween(random(), IndexVersions.INFERENCE_METADATA_FIELDS, IndexVersion.current())
-        );
-    }
-
     @SuppressWarnings({ "unchecked", "rawtypes" })
     public void testFilterNoop() throws Exception {
-        IndexVersion indexVersion = getRandomIndexVersion();
-        ShardBulkInferenceActionFilter filter = createFilter(threadPool, Map.of(), DEFAULT_BATCH_SIZE, indexVersion);
+        boolean useLegacyFormat = randomBoolean();
```

Comment:
Should we use test parameters using `@ParametersFactory` to ensure both legacy and new version test cases are checked? That would help detect regressions earlier

Reply:
Thanks that cleaner than my loop over true/false ;)
I pushed https://github.com/elastic/elasticsearch/pull/119070/commits/83fdc3cc78dc0319d3cf008e0a14adc1d5a83558 to apply this to all tests of the semantic text field.